File: java-sdks/src/main/java/com/suisdks/sui/grpc/JsonUtil.java
package com.suisdks.sui.grpc;

import com.google.gson.Gson;
import com.google.gson.reflect.TypeToken;
import java.lang.reflect.Type;
import java.util.Map;

final class JsonUtil {
    private static final Gson GSON = new Gson();
    private static final Type MAP_TYPE = new TypeToken<Map<String, Object>>() {
    }.getType();

    private JsonUtil() {
    }

    static String toJson(Object value) {
        return GSON.toJson(value);
    }

    static Map<String, Object> fromJsonObject(String json) {
        return GSON.fromJson(json, MAP_TYPE);
    }
}

File: kotlin-sdks/src/main/kotlin/com/suisdks/sui/grpc/JsonUtil.kt
package com.suisdks.sui.grpc

import com.google.gson.Gson
import com.google.gson.reflect.TypeToken

internal object JsonUtil {
    private val gson = Gson()

    fun toJson(value: Any?): String = gson.toJson(value)

    fun fromJsonObject(json: String): Map<String, Any?> {
        val type = object : TypeToken<Map<String, Any?>>() {}.type
        return gson.fromJson(json, type)
    }
}
